enhancements/baremetal/baremetal-provisioning-config.md

```diff
@@ -0,0 +1,211 @@
+---
+title: Baremetal IPI Config Enhancement
+authors:
+  - "@sadasu"
+reviewers:
+  - "@deads2k"
+  - "@hardys"
+  - "@enxebre"
+  - "@squeed"
+  - "@abhinavdahiya"
+  - "@dhellmann"
+
+approvers:
+  - "@abhinavdahiya"
+  - "@enxebre"
+  - "@deads2k"
+
+creation-date: 2019-11-14
+last-updated: 2019-11-26
+status: not implemented
+see-also:
+  - 
+replaces:
+  - https://github.com/openshift/enhancements/pull/90
+superseded-by:
+  - 
+---
+
+# Config required for Baremetal IPI deployments
+
+## Release Signoff Checklist
+
+- [*] Enhancement is `implementable`
+- [*] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift/docs]
+
+## Open Questions
+
+1. Which is a preferred name for the new CR "Metal3ProvisioningController" or
+"Metal3Controller"?
+
+[Closed] Based on review comments it appears that "Metal3ProvisioningController"
+is the preferred name since it makes the contents of the config very clear.
+
+2. There is a possibility that the MCO would also have to refer to this new CR
+created in operator scope. Would it be OK for the MCO to access the CR with this
+(limited) scope?
+
+[Closed] MCO will be able to read config from this new CR if needed.
+
+## Summary
+
+The configuration representing different provisioning network parameters used
+for provisioning baremetal servers need to be made available to the Machine
+API Operator (MAO) in a Config Resource. We had submitted an earlier proposal
```

Comment:
>to be made available to the Machine API Operator (MAO)

This is a baremetal peculiarity, can we specify which components do actually need this? e.g to be made available to the baremetaloperator(something else) which is managed by the mao? 

Is this config really needed at the time the mao defines the manifests which run the baremetal pods, or is this something that can be consumed directly by the baremetaloperator controller at run time?

Reply:
The "baremetal-operator" is terribly named. It's not an operator in the sense that we use that term elsewhere, and we didn't learn how specifically the term "operator" is used before we were far too late in the development to rename it. The baremetal-operator is just a controller, and is part of the metal3 application.

The machine-api-operator is the Operator responsible for managing the configuration of metal3. It will do that by passing values from this new CRD to the application via the Deployment. The components inside metal3 do not know about or consume the CRD.

Reply:
I have tried to address @enxebre 's questions within the doc to provide clarity on when and how the config in the CRs are being consumed.